ledger_device_sdk/src/nbgl.rs

```diff
@@ -162,7 +118,7 @@ unsafe fn settings_callback(token: ::core::ffi::c_int, _index: u8, _page: ::core
         let mut switch_values: [u8; SETTINGS_SIZE] = data.get_ref().clone();
         switch_values[setting_idx] = !switch_values[setting_idx];
         data.update(&switch_values);
-        SWITCH_ARRAY[setting_idx].initState = switch_values[setting_idx] as nbgl_state_t;
+        SWITCH_ARRAY[setting_idx].initState = !SWITCH_ARRAY[setting_idx].initState;
```

Comment:
I think we should leave ` SWITCH_ARRAY[setting_idx].initState = switch_values[setting_idx] as nbgl_state_t;` to make 100% sure there is no way for the displayed switch to have a value different than what is written in NVM

Reply:
done